src/plugins/sensors/gob_sensor.py

```diff
@@ -0,0 +1,77 @@
+"""
+GOB Sensor
+
+The GOB sensor listens to GOB workflow messages
+"""
+import logging
+
+from airflow.exceptions import AirflowException
+from airflow.operators.sensors import BaseSensorOperator
+from airflow.utils.decorators import apply_defaults
+
+from colour import Color
+
+from utils.connection import Connection
+from config.rabbitmq_config import RESULT_QUEUE
+
+
+class GOBSensor(BaseSensorOperator):
+    ui_color = Color("lime").hex
+
+    @apply_defaults
+    def __init__(self, *args, **kwargs):
+        """
+        Initializes the GOB sensor with the GOB-RabbitMQ connection
+        """
+        super().__init__(*args, **kwargs)
+
+        self.connection = Connection()
+
+    def poke(self, context):
+        """
+        Try to read workflow messages for the current task (DAG run_id)
+        :param context: Airflow context
+        :return: final result message for the current task or None if not yet available
+        """
+        # todo force reschedule when no result received
+        result = None
+        for msg in self.connection.consume(RESULT_QUEUE):
+            if msg is None:
+                logging.info("No message yet. Waiting...")
+                break
+
+            if msg['header']['airflow']["run_id"] != context['dag_run'].run_id:
+                logging.info("Skip message for other workflow")
+                continue
+
+            self.connection.ack(msg)
+
+            status = msg.get('status')
+            if status is not None:
```

Comment:
which process sets this status? is it the service self or workflow? do we really need this? Is it not enough that the message appeared in the queue?

Reply:
This requires knowledge of GOB, it's a good question.

As soon as a module starts handling a message (which might be long after it has been send, because of queueing) it reports that it has started handling. At the end it reports that it has finished. In the future the task is even going to report at what % it is. It is very important information for any workflow management tool in a message driven environment.

It is different from ordinary programming where a call starts a process. In GOB a message starts a process and it may take a long time before any GOB component is able to start handling the message. Think "message bus" instead of procedure calls.